actors/eam/src/lib.rs

```diff
@@ -194,6 +209,51 @@ fn resolve_caller(rt: &mut impl Runtime) -> Result<EthAddress, ActorError> {
     })
 }
 
+fn resolve_caller_external(rt: &mut impl Runtime) -> Result<(EthAddress, EthAddress), ActorError> {
+    let caller = rt.message().caller();
+    let caller_id = caller.id().unwrap();
+    if let Some(caller_code_cid) = rt.get_actor_code_cid(&caller_id) {
+        if caller_code_cid == rt.get_code_cid_for_type(Type::Account) {
```

Comment:
I'd invert this. I.e., 

1. Call `resolve_builtin_actor_type`.
2. Check the type.

That way we only have to make one syscall.